sdk/core/loader/boot.cc

```diff
@@ -1267,15 +1267,31 @@ extern "C" SchedulerEntryInfo loader_entry_point(const ImgHdr &imgHdr,
 	              sizeof(void *),
 	              PermissionSet{Permission::Global, Permission::Unseal});
 
-	setSealingKey(imgHdr.allocator(), Allocator);
+	/*
+	 * The token library unseals both static and dynamic objects, sometimes
+	 * either and sometimes with static knowledge of which is expected.  To
+	 * avoid `li; csetaddr; csetbounds` sequences, we give it a separate cap
+	 * for each case.
+	 */
 	setSealingKey(imgHdr.token_library(),
 	              Allocator,
 	              2, // Allocator and StaticToken
```

Comment:
Could we have a `static_assert` here that `StaticToken == (Allocator + 1)`?

Reply:
It's already there, further up:
https://github.com/CHERIoT-Platform/cheriot-rtos/blob/049fb9e328f383171937c32ec1927abd9b61eef1/sdk/core/loader/boot.cc#L155-L158